Example/ScanDemo 2/Pods/CML_SCAN/ScanView.swift

```diff
@@ -0,0 +1,284 @@
+//
+//  ScanView.swift
+//  SCAN
+//
+//  Created by sukhjeet singh sandhu on 27/07/17.
+//  Copyright © 2017 Solcen. All rights reserved.
+//
+
+import UIKit
+import AVFoundation
+import AudioToolbox
+
+// These are the barcode types, those are supported by this Pod.
+public enum Types {
+    case qrCode
+    case dataMatrix
+    case ean8
+    case ean13
+    case code128
+    case code39
+}
+
+// This protocol is responsibe for providing the barcode and type back to the view controller.
+public protocol BarcodeScannerDelegate: class {
+    func didReceiveBarcode(barcode: String?, barcodeType: String?)
```

Comment:
If there is an error in getting the barcode value, we will send nil so that an error can be presented.